src/components/core/geoChart/GeoChartRenderer.tsx

```diff
@@ -114,10 +115,23 @@ export default class GeoChartRenderer extends React.PureComponent<IGeoChartRende
 
         const geoData = getGeoData(buckets, executionResponse.dimensions);
         chart.addSource(DEFAULT_DATA_SOURCE_NAME, createPushpinDataSource(executionResult, geoData));
-        chart.addLayer(
-            createPushpinDataLayer(DEFAULT_DATA_SOURCE_NAME, executionResult, geoData, selectedSegmentItem),
-            "state-label", // pushpin will be rendered under state/county label
-        );
+
+        if (geoData.size !== undefined) {
```

Comment:
this check should be consistent with this line https://github.com/gooddata/gooddata-react-components/pull/1329/files#diff-6b220fe05c6963feb3422220c77bcfa9R132
```
if (!geoData.size) {...}
```